bug: node-pool name max length in gcp is 40 chars

If the generated node-pool name exceeds the 40 char length the process fails with below error. 

1. Either shorten the node-pool name or make it static in default cluster terrafom config
2. Validate the name length before deployment and handle the error.

```
createCluster running with flags
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15a03ca]
goroutine 1 [running]:
github.com/iac-io/myiac/internal/gcp.CreateGCSBucket(0x7ffeefbffa88, 0xf, 0x7ffeefbffa9e, 0x3, 0x17c3180, 0xc0002de960)
        /Users/boraozkan/GolandProjects/myiac/internal/gcp/bucket.go:44 +0x32a
github.com/iac-io/myiac/internal/cluster.InitTerraform(0xc000042a80, 0x40, 0x7ffeefbffa88, 0xf, 0x7ffeefbffa9e, 0x3)
        /Users/boraozkan/GolandProjects/myiac/internal/cluster/cluster.go:34 +0x67
github.com/iac-io/myiac/internal/cluster.CreateCluster(0x7ffeefbffa88, 0xf, 0x7ffeefbffa9e, 0x3, 0x17981f2, 0xe, 0x1, 0x21, 0x0)
        /Users/boraozkan/GolandProjects/myiac/internal/cluster/cluster.go:61 +0x65
github.com/iac-io/myiac/internal/cli.createClusterCmd.func1(0xc0000c8dc0, 0xc000207300, 0xc0002073b0)
        /Users/boraozkan/GolandProjects/myiac/internal/cli/cli.go:299 +0x31a
github.com/urfave/cli.HandleAction(0x167d340, 0x17c20d0, 0xc0000c8dc0, 0xc0000c8dc0, 0x0)
        /Users/boraozkan/go/pkg/mod/github.com/urfave/cli@v1.22.4/app.go:526 +0xfd
github.com/urfave/cli.Command.Run(0x179796a, 0xd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x17adbff, 0x2d, 0x0, ...)
        /Users/boraozkan/go/pkg/mod/github.com/urfave/cli@v1.22.4/command.go:174 +0x58e
github.com/urfave/cli.(*App).Run(0xc0002581c0, 0xc0000320b0, 0xb, 0xb, 0x0, 0x0)
        /Users/boraozkan/go/pkg/mod/github.com/urfave/cli@v1.22.4/app.go:279 +0x7e8
github.com/iac-io/myiac/internal/cli.BuildCli()
        /Users/boraozkan/GolandProjects/myiac/internal/cli/cli.go:66 +0x8d0
main.main()
        /Users/boraozkan/GolandProjects/myiac/cmd/myiac/myiac.go:16 +0x25
```

--- a/internal/cli/create_secret.go
+++ b/internal/cli/create_secret.go
@@ -5,6 +5,7 @@ import (
 	"log"
 	"strings"
 
+	"github.com/iac-io/myiac/internal/cluster"
 	"github.com/iac-io/myiac/internal/gcp"
 	"github.com/iac-io/myiac/internal/secret"
 	"github.com/urfave/cli"

--- a/internal/cluster/cluster.go
+++ b/internal/cluster/cluster.go
@@ -62,13 +62,8 @@ func ApplyTerraform(tp string, tf string) {
 func CreateCluster(provider string, project string, env string, zone string, dryrun bool, key string) error {
 	InitTerraform(tfvarsPath, project, env)
 	if dryrun {
-		log.Printf("Authenticating only due to --dry-run: %v", project+"-"+env)
-		SetupProvider(provider, zone, project+"-"+env, project, key, dryrun)
 		PlanTerraform(tfvarsPath, tfvarsFile)
 	} else {
-		// Set local env kube for local connectivity to new cluster
-		log.Printf("Setting kubectl to work with new cluster: %v", project+"-"+env)
-		SetupProvider(provider, zone, project+"-"+env, project, key, dryrun)
 		ApplyTerraform(tfvarsPath, tfvarsFile)
 	}
 

--- a/internal/terraform/cluster/cluster.tf
+++ b/internal/terraform/cluster/cluster.tf
@@ -4,8 +4,7 @@ locals {
   network_cidr            = "10.254.0.0/16"
 
   cluster_name            = "${var.project}-${var.environment}"
-  applications_pool_name  = "${var.project}-${var.environment}-applications-pool"
-  elasticsearch_pool_name = "${var.project}-${var.environment}-elasticsearch-pool"
+  applications_pool_name  = "default-pool"
 }
 
 provider "google" {
@@ -13,6 +12,7 @@ provider "google" {
   region      = var.cluster_zone
 }
 
+//TODO: Find a way to pass those on run.
 terraform {
   backend "gcs" {
     bucket = "ozzy-playground-tfstate-dev"
@@ -34,8 +34,8 @@ resource "google_container_cluster" "cluster" {
   location = var.cluster_zone
   network  = google_compute_network.gke_network.self_link
   subnetwork = google_compute_subnetwork.gke_subnet.self_link
-  min_master_version = "1.17.14-gke.1600"
-  node_version = "1.17.14-gke.1600"
+  min_master_version = var.k8s_master_version
+  node_version = var.k8s_node_pool_version
 
   remove_default_node_pool = true
   logging_service          = "logging.googleapis.com/kubernetes"
@@ -91,43 +91,7 @@ resource "google_container_node_pool" "applications_node_pool" {
   }
 }
 
-//resource "google_container_node_pool" "elasticsearch_node_pool" {
-//  provider           = google
-//  project            = var.project
-//  name               = local.elasticsearch_pool_name
-//  location           = google_container_cluster.cluster.location
-//  cluster            = google_container_cluster.cluster.name
-//  initial_node_count = 1
-//
-//  management {
-//    auto_repair  = true
-//    auto_upgrade = true
-//  }
-//
-//  autoscaling {
-//    # Minimum number of nodes in the NodePool. Must be >=0 and <= max_node_count.
-//    min_node_count = 1
-//
-//    # Maximum number of nodes in the NodePool. Must be >= min_node_count.
-//    max_node_count = var.elasticsearch_max_node_count
-//  }
-//
-//  node_config {
-//    preemptible  = true
-//    machine_type = var.elasticsearch_machine_type
-//    disk_size_gb = 20
-//
-//    oauth_scopes = [
-//      "https://www.googleapis.com/auth/compute",
-//      "https://www.googleapis.com/auth/devstorage.read_only",
-//      "https://www.googleapis.com/auth/logging.write",
-//      "https://www.googleapis.com/auth/monitoring",
-//    ]
-//  }
-//}
-
 # Firewall rule for NodePort
-
 resource "google_compute_subnetwork" "gke_subnet" {
   project       = var.project
   name          = "${var.project}-${var.environment}-gke-subnet"

--- a/internal/terraform/cluster/cluster.tfvars
+++ b/internal/terraform/cluster/cluster.tfvars
@@ -4,8 +4,8 @@ cluster_zone                 = "europe-west2-b"
 cluster_state_bucket         = "ozzy-playground-tfstate-dev"
 state_bucket_prefix          = "terraform/state/cluster"
 applications_machine_type    = "n1-standard-2"
-elasticsearch_machine_type   = "n1-standard-1"
 applications_max_node_count  = 3
-elasticsearch_max_node_count = 1
+k8s_master_version           = "1.17.15-gke.800"
+k8s_node_pool_version        = "1.17.15-gke.800"
 //terraform apply -var-file="cluster.tfvars" 
 //https://github.com/jetstack/terraform-google-gke-cluster/blob/master/example/main.tf

--- a/internal/terraform/cluster/variables.tf
+++ b/internal/terraform/cluster/variables.tf
@@ -22,16 +22,14 @@ variable "applications_machine_type" {
   type = string
 }
 
-variable "elasticsearch_machine_type" {
-  type = string
-}
-
 variable "applications_max_node_count" {
   type = string
 }
 
-variable "elasticsearch_max_node_count" {
-  type = number
+variable "k8s_master_version" {
+  type = string
 }
 
-
+variable "k8s_node_pool_version" {
+  type = string
+}

